### PR TITLE
Navkiran feature 4 return an item

### PR DESCRIPTION
**Pull request by NavGill12**
I added all the return files to make a working return function. I also made some changes to AddScreen and the Main2()

Add Screen
- Changes the access modifiers to protected to allow access by the child class
- Added a constructor that takes in screens, to create an object that doesn't need an additional controller parameter
Main Screen
-Added closing functionality to the panel 
- Initialised required objects for return class
The changes made to the main2 are only temporary until we have a final Builder for that functionality.
I haven't added documentation yet, but will do so shortly

### CODE REVIEW

**Comment by tonyxieuoft** [comment]
For some reason Git isn't letting me comment on specific lines of code:

1) Your addition of the InputData for the change history use case is great! I really like how it removed the bloater that was there before (with the long parameter list).

2) We might want to consider passing in the necessary classes for the history use case to work (ex. gateway) into our own classes during construction for dependency inversion. If it's too cumbersome, it's ok if we don't.

File: src/main/java/use_cases/change_history_use_case/ChangeHistoryData.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package use_cases.change_history_use_case;
+
+import entities.Item;
+import use_cases.arr.ARRInputData;
+
+public class ChangeHistoryData {
```
Comment (wuhanyi70) [inline src/main/java/use_cases/change_history_use_case/ChangeHistoryData.java:6]: I think i need to change my changehistorytest according to this.

File: src/main/java/use_cases/change_history_use_case/ChangeHistory.java [hunk 2]
```diff
@@ -24,16 +24,16 @@ public class ChangeHistory {
 
 //GET all the history data we need when constructing ChangeHistory object
 
-    public ChangeHistory(String name, String action, String item, String quantity, String serinum)  {
+    public ChangeHistory(ChangeHistoryData data)  {
         //get current time and save it as a String arribute
```
Comment (wuhanyi70) [inline src/main/java/use_cases/change_history_use_case/ChangeHistory.java:28]: Adding a data structure making this function clearer

File: src/main/java/interface_adaptors/arr/RemovePresenter.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import java.awt.*;
+
+public class RemovePresenter implements ARROutputBoundary {
+    /**
+     * A presenter for the Remove functionality.
+     *
+     * It is a part of the MVC model functionality adopted for the Add, Return and Remove functionalities for
+     * updating the quantity of the item. Calls are made to the ARROutputBoundary .
+     *
+     */
```
Comment (wuhanyi70) [inline src/main/java/interface_adaptors/arr/RemovePresenter.java:16]: screen set up

File: src/main/java/use_cases/arr/Add.java [hunk 4]
```diff
@@ -17,7 +21,8 @@ public Add(ARROutputBoundary presenter){
      */
     public void changeItemQuantity(ARRInputData data){
 
-        if (data.getQuantity() < 0){
+        if (data.getQuantity() < 0 && !(this instanceof RemoveUseCase)) // To allow negative quantity values only when the item is being removed
+        {
```
Comment (Robinrswanson) [inline src/main/java/use_cases/arr/Add.java:26]: Initial if statement might be redundant and can be removed. Keep 2nd if statement.

**Review by Robinrswanson — COMMENTED** [review]
Looks good and adheres to clean architecture. Just the one change to consider in the Add class. Good Work.

**Pull request merged**